skbio/io/format/genbank.py

```diff
@@ -0,0 +1,468 @@
+    '''
+    res = {}
+    section_splitter = yield_section(lambda x: not x.startswith(' ' * 11),
+                                     skip_blanks=True, strip=False)
+    # SOURCE line is not informative; skip it
+    _, organism = list(section_splitter(lines))
+
+    res['ORGANISM'] = organism[0].split(None, 1)[1].strip()
+    res['taxonomy'] = ' '.join([i.strip() for i in organism[1:]])
+    return res
+
+
+def _parse_features(lines, length):
+    '''Parse FEATURES field.
+    '''
+    features = []
+    positional_metadata = []
+    # skip the 1st FEATURES line
+    if lines[0].startswith('FEATURES'):
+        lines = lines[1:]
+    # magic number 20: the lines following header of each feature
+    # are at least indented with 20 spaces.
+    n = 20
+    section_splitter = yield_section(
+        lambda x: not x.startswith(' ' * n),
+        skip_blanks=True, strip=False)
+    for i, section in enumerate(section_splitter(lines)):
+        feature, pmd = _parse_single_feature(section, length, i)
+        features.append(feature)
+        positional_metadata.append(pmd)
+    return features, positional_metadata
+
+
+def _parse_single_feature(lines, length, index):
+    '''Parse a feature.
+
+    Returns
+    -------
+    tuple
+        Tuple of a dict of `metadata` and a pandas.Series of
+        `positional_metadata` for the feature.
+
+    '''
+    feature = dict()
+    feature['index_'] = index
+    # each component of a feature starts with '/', except the 1st
+    # component of location.
+    section_splitter = yield_section(
+        lambda x: x.startswith('/'), strip=True)
+    first = True
+    for section in section_splitter(lines):
+        if first:
+            # first section is the Location string
+            first = False
+            type, location = _parse_section_default(
+                section, join_delimitor='', return_label=True)
+            feature['type_'] = type
+            feature['location'] = location
+            loc, loc_pmd = _parse_loc_str(location, length)
+            feature.update(loc)
+        else:
+            # following sections are Qualifiers
+            k, v = _parse_section_default(
+                section, label_delimitor='=',
+                join_delimitor=' ', return_label=True)
+            k = k[1:]
+            # strip the quotes if it is quoted.
+            # v could be empty
+            if v and v[0] == '"' and v[-1] == '"':
+                v = v[1:-1]
+            # some Qualifiers can appear multiple times
+            if k in feature:
+                feature[k] = [feature[k]]
+                feature[k].append(v)
+            else:
+                feature[k] = v
+    return feature, loc_pmd
+
+
+def _parse_loc_str(loc_str, length):
+    '''Parse location string.
+
+    Warning: This converts coordinates to 0-based from 1-based as
+    in Genbank format.
+    '''
+    pmd = np.zeros(length, dtype=bool)
+    res = {'rc_': False,
+           'left_partial_': False,
```

Comment:
not dynamic. they are finite for the time being, but since genbank format is changing slightly, it may need expansion. The reason I prefer that is 2 folds: 1. users can easily spot them when they look at `metadata`; 2. it can be generalized as a rule for parsers of other complicated formats. I think it is better than setting up another lookup table...